sky/serve/serve_utils.py

```diff
@@ -55,6 +55,70 @@ def get_num_service_threshold():
     return system_memory_gb // constants.CONTROLLER_MEMORY_USAGE_GB
 
 
+def _read_last_n_lines(file_handle: typing.TextIO, n: int) -> List[str]:
+    """Read the last N lines from a file without loading the entire file.
+    This reverse reads the file in chunks from the end,
+    which is much more memory-efficient than reading the entire file.
+    Args:
+        file_handle: Open file handle
+        n: Number of lines to read from the end
+    Returns:
+        List of the last N lines (with newlines preserved)
+    """
+    file_handle.seek(0, 2)  # Seek to end
```

Comment:
```suggestion
    file_handle.seek(offset=0, whence=2)  # Seek to end
```
nit: for literal arguments, lets use kwargs format

Reply:
TextIOWrapper.seek() and TextIOWrapper.read() doesn't allow keyword arguments